packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,153 @@
+
+### How it works
+
+For properties, the wrapper interrogates the web component class to discover its available properties. Then any React props passed with property names are set on the custom element as properties and not attributes.
+
+For events, `createComponent()` accepts a mapping of React event prop names to events fired by the custom element. For example passing `{onFoo: 'foo'}` means a function passed via a prop named `onFoo` will be called when the custom element fires the `foo` event with the event as an argument.
+
+### Usage
+
+Import `React`, a custom element class, and `createComponent`.
+
+```js
+import React from 'react';
+import {createComponent} from '@lit-labs/react';
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onActivate: 'activate',
+    onChange: 'change',
+  },
+});
+```
+
+After defining the React component, you can use it just as you would any other React component.
+
+```jsx
+<MyElementComponent
+  active={isActive}
+  onActivate={(e) => setIsActive(e.active)}
+/>
+```
+
+#### Typescript
+
+Event callback types can be refined by type casting with `EventName`. The type cast helps `createComponent` correlate typed callbacks to property names in the event property map.
+
+Non-casted event names will fallback to an event type of `Event`.
+
+```ts
+import type {EventName} from '@lit-labs/react';
+
+import React from 'react';
+import {createComponent} from '@lit-labs/react';
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onClick: 'pointerdown' as EventName<PointerEvent>,
+    onChange: 'input',
+  },
+});
+```
+
+Event callbacks will match their type cast. In the example below, a `PointerEvent` is expected in the `onClick` callback.
+
+```tsx
+<MyElementComponent
+  onClick={(e: PointerEvent) => {
+    console.log('DOM PointerEvent called!');
+  }}
+  onChange={(e: Event) => {
+    console.log(e);
+  }}
+/>
+```
+
+{% aside "info" "no-header" %}
+This type casting is not associated to any component property. Be careful to use the corresponding type dispatched or bubbled from the webcomponent. Incorrect types might result in additional properties, missing properties, or properties of the wrong type.
+{% endaside %}
+
+## useController
+
+Reactive controllers allow developers to hook in to a component's lifecycle to bundle
+together state and behavior related to a feature. They are similar to React
+hooks in the user cases and capabilities, but are plain JavaScript objects
+instead of functions with hidden state.
+
+`useController()` is a React hook that creates and stores a reactive controller
```

Comment:
This is a bit more how it works that why you'd use it. I think saying that `useController()` let you make hooks out of controllers so you can share "headless" state and behavior across web components and frameworks is key.